src/unit.rs

```diff
@@ -1592,11 +1600,13 @@ fn length_uleb128_value<R: Reader>(input: &mut R) -> Result<R> {
     input.split(len)
 }
 
-fn parse_attribute<'unit, R: Reader>(
+fn parse_attribute<'unit, 'abbrev, R: Reader>(
     input: &mut R,
     unit: &'unit UnitHeader<R, R::Offset>,
-    spec: AttributeSpecification,
-) -> Result<Attribute<R>> {
+    mut specs: &'abbrev [AttributeSpecification],
+) -> Result<(Attribute<R>, &'abbrev [AttributeSpecification])> {
+    let spec = specs[0];
+    specs = &specs[1..];
```

Comment:
This change isn't needed anymore, and it's part of what I thought would affect performance, but reverting it didn't help, so going to leave any cleanup here for future performance work.